src/plugins/bfetch/server/plugin.ts

```diff
@@ -209,21 +210,16 @@ export class BfetchServerPlugin
       const handlerInstance = handler(request);
       return {
         getResponseStream: ({ batch }) => {
-          const subject = new Subject<BatchResponseItem<BatchItemResult, E>>();
-          let cnt = batch.length;
-          batch.forEach(async (batchItem, id) => {
-            try {
-              const result = await handlerInstance.onBatchItem(batchItem);
-              subject.next({ id, result });
-            } catch (err) {
-              const error = normalizeError<E>(err);
-              subject.next({ id, error });
-            } finally {
-              cnt--;
-              if (!cnt) subject.complete();
-            }
-          });
-          return subject;
+          return Rx.from(batch).pipe(
+            mergeMap(async (batchItem, id) => {
```

Comment:
In the original implementation `handlerInstance.onBatchItem()` was called immediately, and didn't wait for the subject to be subscribed. In my eyes this was likely unintended and a race condition (results could be emitted before the subject is subscribed too) which is removed by the new implementation. Please let me know if the original implementation was intended.